### PR TITLE
[Main page] Show project loading spinner while project loads

### LINKED ISSUE #296: [Project] Hide project page until loaded
- [ ] Consider using Resolvers to load project before route enables components.
- [ ] Replace async reference to project, ACLs, layer config, etc. with simpler sync reference

### PR DESCRIPTION
**Pull request by gino-m**
Closes #296

### CODE REVIEW

File: web-ng/src/app/components/main-page-container/main-page-container.module.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { CommonModule } from '@angular/common';
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { MainPageContainerComponent } from './main-page-container.component';
+import { MainPageModule } from './../main-page/main-page.module';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { NgModule } from '@angular/core';
+import { RouterModule } from '@angular/router';
+
+@NgModule({
+  entryComponents: [],
```
Comment (parulraheja98) [inline web-ng/src/app/components/main-page-container/main-page-container.module.ts:26]: Entry components can be safely removed and is expected to be removed in future angular versions.
https://angular.io/guide/deprecations

File: web-ng/src/app/components/main-page/main-page.module.ts [hunk 2]
```diff
@@ -23,25 +24,27 @@ import { MapModule } from '../map/map.module';
...
 import { MatSidenavModule } from '@angular/material/sidenav';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 import { NgModule } from '@angular/core';
-import { RouterModule } from '@angular/router';
 import { SidePanelModule } from '../side-panel/side-panel.module';
 import { ProjectHeaderModule } from '../project-header/project-header.module';
 import { MatIconModule } from '@angular/material/icon';
 
 @NgModule({
   entryComponents: [LayerDialogComponent],
```
Comment (parulraheja98) [inline web-ng/src/app/components/main-page/main-page.module.ts:34]: entry components can be removed

**Review by parulraheja98 — DISMISSED** [review]
LGTM with minor suggestions on deprecations.

File: web-ng/src/app/components/main-page-container/main-page-container.module.ts (hunk 1 above)
Comment (gino-m) [reply web-ng/src/app/components/main-page-container/main-page-container.module.ts:26]: Good to know, thanks for catching this!

**Review by parulraheja98 — APPROVED** [review]
LGTM 🚢

**Pull request merged**